base/base.jl

```diff
@@ -82,7 +82,7 @@ finalize(o::ANY) = ccall(:jl_finalize, Void, (Any,), o)
 gc(full::Bool=true) = ccall(:jl_gc_collect, Void, (Cint,), full)
 gc_enable(on::Bool) = ccall(:jl_gc_enable, Cint, (Cint,), on)!=0
 
-bytestring(str::ByteString) = str
+bytestring(str::String) = str
```

Comment:
Yep, understood. Definitely excited for this and I'll try to take some more time to review/test things out to help out.